### ISSUE TITLE
Zeros on first tensor concatenation op

### ISSUE THREAD

**Issue by Litee** [tensorflow/tfjs#381]
#### TensorFlow.js version

0.11.4

#### Browser version

Chrome 66.0.3359.181
Windows 10

#### Describe the problem or feature request

model.predict() results contain a range of all zeros - it looks like the first concatenate operation with result longer than 8192 produces zeros. Here is what I see when I look at results (see minimal sample code below):

...
Slice:  818 Tensor
     [[0.039277, 0.0680138, 0.0480547, 0.0202137, 0.0317248, 0.0945501, 0.0599228, 0.0707796, 0.4984992, 0.0689642],]
Slice:  819 Tensor
     [[0.0227662, 0.2433799, 0, 0, 0, 0, 0, 0, 0, 0],]
Slice:  820 Tensor
     [[0, 0, 0, 0, 0, 0, 0, 0, 0, 0],]
...
Slice:  1630 Tensor
     [[0, 0, 0, 0, 0, 0, 0, 0, 0, 0],]
Slice:  1631 Tensor
     [[0, 0, 0, 0, 0, 0, 0, 0, 0, 0],]
Slice:  1632 Tensor
     [[0.0360319, 0.2653748, 0.0344116, 0.031091, 0.0786578, 0.0866721, 0.054291, 0.0158215, 0.3704009, 0.0272476],]
Slice:  1633 Tensor
     [[0.01593, 0.1220765, 0.028846, 0.0361838, 0.0279004, 0.0885607, 0.0406828, 0.0039507, 0.5233707, 0.1124983],]
...

#### Code to reproduce the bug / link to feature request

    const model = tf.sequential();

    model.add(tf.layers.dense({
      inputShape: [28 * 28],
      units: 10,
      kernelInitializer: 'varianceScaling',
      activation: 'softmax'
    }));

    model.compile({
      optimizer: 'sgd',
      loss: 'categoricalCrossentropy',
      metrics: ['accuracy'],
    });

    // No training

    tf.tidy(() => {
      console.log('Generating prediction data...');
      const toPredict = tf.randomUniform([1700, 28 * 28]);
      console.log('Predicting...');
      const prediction = model.predict(toPredict);
      for (let i = 0; i < prediction.shape[0]; i++) {
        console.log('Slice: ', i, prediction.slice(i, 1).toString());
      }
    });

**Comment by Litee** [comment]
Another simple way to reproduce is to increase the number of predictions in MNIST example from 100 to 1000 and run the script. If you have this problem on your machine you will see many "0" predictions:

![image](https://user-images.githubusercontent.com/400234/41054924-77d4cd6a-69c8-11e8-81d7-26356c12a0a9.png)

**Comment by Litee** [comment]
OK, it seems to have something to do with concatenation shader and MAX_TEXTURE_SIZE. My default MAX_TEXTURE_SIZE is 16384, but if I fake it to 8192 in code problem disappears. Unfortunately I have zero experience with shaders, so maybe someone can can help? Shader code:

      precision highp float;
      precision highp int;
      varying vec2 resultUV;
      const vec2 halfCR = vec2(0.5, 0.5);

      bool isNaN(float val) {
        float v1 = val * val;
        float v2 = val * val;
        return v1 == v2 ? false : true;
      }

      bool hasNaN(vec4 values) {
        vec4 v1 = values * values;
        vec4 v2 = values * values;
        return any(notEqual(v1, v2));
      }

      float getNaN(vec4 values) {
        return dot(vec4(1), values);
      }

      int round(float value) {
        return int(floor(value + 0.5));
      }

      int imod(int x, int y) {
        return x - y * (x / y);
      }

      //Based on the work of Dave Hoskins
      //https://www.shadertoy.com/view/4djSRW
      #define HASHSCALE1 443.8975
      float random(float seed){
        vec2 p = resultUV * seed;
        vec3 p3  = fract(vec3(p.xyx) * HASHSCALE1);
        p3 += dot(p3, p3.yzx + 19.19);
        return fract((p3.x + p3.y) * p3.z);
      }

      
    vec2 UVfrom1D(int texNumR, int texNumC, int index) {
      int texR = index / texNumC;
      int texC = index - texR * texNumC;
      return (vec2(texC, texR) + halfCR) / vec2(texNumC, texNumR);
    }

      
    vec2 UVfrom2D(int texNumR, int texNumC, int numC, int row, int col) {
      int index = row * numC + col;
      int texR = index / texNumC;
      int texC = index - texR * texNumC;
      return (vec2(texC, texR) + halfCR) / vec2(texNumC, texNumR);
    }

      
    vec2 UVfrom3D(int texNumR, int texNumC, int stride0,
        int stride1, int row, int col, int depth) {
      // Explicitly use integer operations as dot() only works on floats.
      int index = row * stride0 + col * stride1 + depth;
      int texR = index / texNumC;
      int texC = index - texR * texNumC;
      return (vec2(texC, texR) + halfCR) / vec2(texNumC, texNumR);
    }

      
    vec2 UVfrom4D(int texNumR, int texNumC, int stride0,
        int stride1, int stride2, int row, int col, int depth,
        int depth2) {
      // Explicitly use integer operations as dot() only works on floats.
      int index = row * stride0 + col * stride1 + depth * stride2 + depth2;
      int texR = index / texNumC;
      int texC = index - texR * texNumC;
      return (vec2(texC, texR) + halfCR) / vec2(texNumC, texNumR);
    }



      float sampleTexture(sampler2D textureSampler, vec2 uv) {
        return texture2D(textureSampler, uv).r;
      }


      void setOutput(float val) {
        gl_FragColor = vec4(val, 0, 0, 0);
      }

    uniform sampler2D A;
    uniform sampler2D B;

          ivec2 getOutputCoords() {
            return ivec2(resultUV.yx * vec2(1, 8320));
          }
        

          float getAFlat(int index) {
            vec2 uv = vec2((float(index) + 0.5) / 8000.0, 0.5);
            return sampleTexture(A, uv);
          }
        
        float getA(int row, int col) {
          vec2 uv = (vec2(col, row) + halfCR) / vec2(8000.0, 1.0);
          return sampleTexture(A, uv);
        }
      

        float getBFlat(int index) {
          vec2 uv = UVfrom1D(32, 10, index);
          return sampleTexture(B, uv);
        }
      
          
        float getB(int index) {
          return getBFlat(index);
        }
      
          float getB(int row, int col) {
            return getB(col);
          }
        

          void main() {
            ivec2 coords = getOutputCoords();
            int yR = coords.x;
            int yC = coords.y;

            float value = 0.0;
            if (yC < 8000) {
              value = getA(yR, yC);
            } else {
              yC -= 8000;
              value = getB(yR, yC);
            }

            setOutput(value);
          }

**Comment by dsmilkov** [comment]
We got a windows machine with Intel GPU and we are unable to reproduce at problem at `tfj@1.0.2`. Feel free to reopen if you still run into this problem.

**Comment by Litee** [comment]
Thanks, I cannot reproduce the problem anymore.
